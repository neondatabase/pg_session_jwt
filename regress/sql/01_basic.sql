-- Basic functionality tests for pg_session_jwt

-- Test auth.init() function
SELECT auth.init();

-- Test creating a session with a valid JWT
-- In real tests, we would use a function to generate a valid JWT
DO $$
DECLARE
    -- This is a placeholder value that should be replaced with a real JWT at test time
    test_jwt TEXT := 'PLACEHOLDER.JWT.VALUE';
BEGIN
    PERFORM auth.jwt_session_init(test_jwt);
END;
$$;

-- Test auth.session() function
SELECT auth.session() IS NOT NULL AS session_exists;

-- Test auth.user_id() function
SELECT auth.user_id() AS user_id;
