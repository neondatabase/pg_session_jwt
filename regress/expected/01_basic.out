-- Basic functionality tests for pg_session_jwt

-- Test auth.init() function
SELECT auth.init();
 init 
------
 
(1 row)

-- Test creating a session with a valid JWT
-- In real tests, we would use a function to generate a valid JWT
DO $$
DECLARE
    -- This is a placeholder value that should be replaced with a real JWT at test time
    test_jwt TEXT := 'PLACEHOLDER.JWT.VALUE';
BEGIN
    PERFORM auth.jwt_session_init(test_jwt);
END;
$$;
DO

-- Test auth.session() function
SELECT auth.session() IS NOT NULL AS session_exists;
 session_exists 
---------------
 t
(1 row)

-- Test auth.user_id() function
SELECT auth.user_id() AS user_id;
  user_id   
------------
 test_user
(1 row)
